STX.REST.RESTFulSense.Clients/Services/Foundations/HttpExchanges/HttpExchangeService.Validations.Headers.cs

```diff
@@ -0,0 +1,126 @@
+
+using System.Linq;
+using STX.REST.RESTFulSense.Clients.Models.Services.HttpExchanges;
+using STX.REST.RESTFulSense.Clients.Models.Services.HttpExchanges.Exceptions;
+using STX.REST.RESTFulSense.Clients.Models.Services.HttpExchanges.Headers;
+
+namespace STX.REST.RESTFulSense.Clients.Services.Foundations.HttpExchanges
+{
+    internal partial class HttpExchangeService
+    {
+        private static dynamic IsInvalidAcceptHeader(MediaTypeHeader[] mediaTypeHeaders) => new
+        {
+            Condition =
+                mediaTypeHeaders is not null
+                    && (mediaTypeHeaders.Any(header => header is null)
+                        || mediaTypeHeaders.Any(IsInvalidMediaTypeHeader)),
+
+            Message = "Accept header has invalid configuration, fix errors and try again."
+        };
+
+        private static dynamic IsInvalidAcceptCharsetHeader(StringWithQualityHeader[] stringWithQualityHeaders) => new
+        {
+            Condition =
+                stringWithQualityHeaders is not null
+                    && (stringWithQualityHeaders.Any(header => header is null)
+                        || stringWithQualityHeaders.Any(IaInvalidStringWithQualityHeader)),
+
+            Message = "Accept Charset header has invalid configuration, fix errors and try again."
+        };
+
+        private static dynamic IsInvalidAcceptEncodingHeader(StringWithQualityHeader[] stringWithQualityHeaders) => new
+        {
+            Condition =
+                    stringWithQualityHeaders is not null
+                        && (stringWithQualityHeaders.Any(header => header is null)
+                            || stringWithQualityHeaders.Any(IaInvalidStringWithQualityHeader)),
+
+            Message = "Accept Encoding header has invalid configuration, fix errors and try again."
+        };
+
+        private static dynamic IsInvalidAcceptLanguageHeader(
+            StringWithQualityHeader[] stringWithQualityHeaders) => new
+            {
+                Condition =
+                stringWithQualityHeaders is not null
+                    && stringWithQualityHeaders.Any(IaInvalidStringWithQualityHeader),
+
+                Message = "Accept Language header has invalid configuration, fix errors and try again."
+            };
+
+        private static bool IsInvalidMediaTypeHeader(MediaTypeHeader header)
+        {
+            return header is not null
+                && (string.IsNullOrWhiteSpace(header.MediaType)
+                    || string.IsNullOrWhiteSpace(header.CharSet));
+        }
+
+        private static bool IaInvalidStringWithQualityHeader(StringWithQualityHeader header)
+        {
+            return header is not null
+                && (string.IsNullOrWhiteSpace(header.Value)
+                    || (header.Quality is not null
+                        && !IsDoubleBetweenZeroAndOne(header.Quality)));
+        }
+
+        private static bool IsDoubleBetweenZeroAndOne(double? value)
```

Comment:
double is not optional here as you said your "header.Quality is not null" is managed inside "IaInvalidStringWithQualityHeader"

"IaInvalidStringWithQualityHeader" has wrong spelling.